### PR TITLE
Some Bug fixes

### PR DESCRIPTION
**Pull request by ghettopato**
fixes magnet link support
fixes a RC when the torrent file gets picked up too early
now downloads all links and not just the first one

### CODE REVIEW

**Comment by ghettopato** [comment]
improved the extension extraction from files a bit.

File: lib/watchers/real-debrid/index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const debug = require('debug')('patbrid:watchers:real-debrid')
 const RealDebridClient = require('real-debrid-api')
 const RealDebridTorrent = require('./torrent')
 const fs = require('fs')
+const path = require('path');
```
Comment (mgoodings) [inline lib/watchers/real-debrid/index.js]: remove this semicolon

File: lib/downloaders/aria2/index.js [hunk 2]
```diff
@@ -24,16 +24,19 @@ class Aria2 {
...
+        params: [`token:${this.secret}`, [link]],
         id: this.id++
       }, (err, res) => {
+        if(err)
+          console.error('[!] adding download failed', err) // not printing
         !err ? resolve(res) : reject(err)
       })
-    })
+    }))
+    return Promise.all(promises)
```
Comment (mgoodings) [inline lib/downloaders/aria2/index.js:42]: add empty new line

File: index.js [hunk 3]
```diff
@@ -37,8 +37,8 @@ const watcher = new RealDebridWatcher(REAL_DEBRID_API_KEY, downloader.download)
 // Watch for new torrent files
 console.log(`[+] Watching '${WATCH_DIR}' for new torrents`)
 
-chokidar.watch(`${WATCH_DIR}/*.torrent`)
-  .on('add', path => watcher.addTorrent(path))
+chokidar.watch([`${WATCH_DIR}/*.magnet`,`${WATCH_DIR}/*.torrent`],{awaitWriteFinish: true})
```
Comment (mgoodings) [inline index.js]: tidy up formatting and added ignore for dot files

File: lib/watchers/real-debrid/index.js [hunk 4]
```diff
@@ -12,6 +13,17 @@ class RealDebridWatcher {
     this.watchList = []
   }
 
+  addFile(file) {
+    debug('addFile', file)
+
+    let fileext = path.extname(file)
```
Comment (mgoodings) [inline lib/watchers/real-debrid/index.js]: use const

File: lib/watchers/real-debrid/index.js [hunk 5]
```diff
@@ -12,6 +13,17 @@ class RealDebridWatcher {
...
+
+    if(fileext.toLowerCase() === '.magnet')
```
Comment (mgoodings) [inline lib/watchers/real-debrid/index.js]: ensure correct spacing and always use {}

File: lib/watchers/real-debrid/index.js (hunk 1 above)
Comment (mgoodings) [reply lib/watchers/real-debrid/index.js]: also please put nodejs requires at the top of the file

**Pull request merged**